std/range/package.d

```diff
@@ -1228,6 +1231,382 @@ unittest
     auto c = chain(a, b);
 }
 
+
+/**
+    Choose one of multiple ranges at runtime.
+
+    The ranges may be different, but they must have the same element type. The
+    result is a range that offers the $(D front), $(D popFront), and $(D
+    empty) primitives. If all input ranges offer random access and $(D
+    length), $(D choose) offers them as well.
+
+    Params:
+        index = which range to choose, must be less than the number of ranges
+        rs = tuple of 1 or more ranges
+
+    Returns:
+        The indexed range. If rs consists of only one range,
+        the return type is an alias of that range's type.
+ */
+auto choose(Ranges...)(size_t index, Ranges rs)
```

Comment:
I thought about that, but `if(c) e1 else e2` has the e1 and e2 reversed from the way they are if there are n arguments. I worry that the specialization would be very confusing.

Also, I used `size_t` for the index to emphasize its commonality with indexing an array. Calculations of such an index will likely be done with size_t's, and then narrowing them to a uint would require the user to insert an ugly cast.